src/main/java/software/amazon/smithy/lsp/Main.java

```diff
@@ -61,21 +65,28 @@ public static void main(String[] args) {
         Socket socket = null;
         InputStream in;
         OutputStream out;
-
+        List<String> argList = Arrays.asList(args);
         try {
-            String port = args[0];
+            Optional<Exception> launchFailure;
+            String port = getOrDefault(argList, 0, "0");
+            String type = getOrDefault(argList, 1, null);
```

Comment:
to avoid order sensitivity it may be worth taking a dependency on https://github.com/argparse4j/argparse4j or similar

Reply:
Yeah, I didn't really want to make this a full blown CLI with argparse, but maybe something we should do in the future if we decide to add more. Great suggestion.

I also just noticed this package doesn't have a great readme, so I'll add some details there (and include the ordering).